### PR TITLE
Add servicelifecycle to spec index

### PR DESCRIPTION
**Pull request by weshaggard**
@mario-guerra step one I've added the SerivceLifeCycle to the spec index. This also adds the RPSaaSMaster indexing and pushed that file to blob storage. I still need to figure out a good way to add the suppression data.

### CODE REVIEW

File: eng/scripts/Update-Spec-Versions.ps1 [hunk 1]
```diff
@@ -14,6 +15,16 @@ function GetJsonFiles($relSpecPath)
   return Get-ChildItem "specification/$relSpecPath" *.json
 }
 
+function GetServiceLifeCycle($allSpecs, $serviceFamily)
+{
+  $stableSpecs = $allSpecs.Where({ $_.ServiceFamily -eq $serviceFamily -and $_.VersionType -eq "stable" -and $_.IsTypeSpec -ne "True"})
+
+  if ($stableSpecs.Count -eq 0) {
+    return "Brownfield"
```
Comment (mario-guerra) [inline eng/scripts/Update-Spec-Versions.ps1:23]: If there are no stable specs, doesn't that mean the service is greenfield?

Comment (weshaggard) [reply eng/scripts/Update-Spec-Versions.ps1:23]: Yes I inverted that. I'm also still a little unsure of this check as service family in this case is the root folder and there might be a lot of disjointed services. Think https://github.com/Azure/azure-rest-api-specs/tree/main/specification/ai where there are multiple SDKs should everything in there be considered brownfield?

Comment (weshaggard) [reply eng/scripts/Update-Spec-Versions.ps1:23]: It looks like @mikeharder includes the full path in the existing typespec requirement check https://github.com/Azure/azure-rest-api-specs/blob/main/eng/scripts/TypeSpec-Requirement.ps1#L140

Comment (weshaggard) [reply eng/scripts/Update-Spec-Versions.ps1]: I've added ResourcePath as part of the check for green/brown field. So it has to match service and resourcepath.

**Pull request merged**